src/components/NcTextArea/NcTextArea.vue

```diff
@@ -71,226 +71,192 @@ export default {
 ```
 </docs>
 
+<script setup lang="ts">
+import type { VueClassType } from '../../utils/VueTypes.ts'
+
+import { mdiAlertCircle, mdiCheck } from '@mdi/js'
+import { computed, useAttrs, useTemplateRef, watch } from 'vue'
+import { createElementId } from '../../utils/createElementId.ts'
+import NcIconSvgWrapper from '../NcIconSvgWrapper/NcIconSvgWrapper.vue'
+import logger from '../../utils/logger.ts'
+
+defineOptions({ inheritAttrs: false })
+
+defineExpose({
+	focus,
+	select,
+})
+
+const props = withDefaults(defineProps<{
+	/**
+	 * The input label, always provide one for accessibility purposes.
+	 * This will also be used as a placeholder unless the placeholder
+	 * prop is populated with a different string.
+	 */
+	label?: string
+
+	/**
+	 * Pass in true if you want to use an external label. This is useful
+	 * if you need a label that looks different from the one provided by
+	 * this component
+	 */
+	labelOutside?: boolean
+
+	/**
+	 * The placeholder of the input. This defaults as the string that's
+	 * passed into the label prop. In order to remove the placeholder,
+	 * pass in an empty string.
+	 */
+	placeholder?: string
+
+	/**
+	 * Toggles the success state of the component.
+	 * Adds a checkmark icon.
+	 */
+	success?: boolean
+
+	/**
+	 * Toggles the error state of the component.
+	 * Adds an error icon.
+	 */
+	error?: boolean
+
+	/**
+	 * Additional helper text message
+	 *
+	 * This will be displayed beneath the input field. In case the field is
+	 * also marked as having an error, the text will be displayed in red.
+	 */
+	helperText?: string
+
+	/**
+	 * Disable the text area
+	 */
+	disabled?: boolean
+
+	/**
+	 * Class to add to the `<textarea>` element.
+	 */
+	inputClass?: VueClassType
+
+	/**
+	 * The resize CSS property sets whether an element is resizable, and if
+	 * so, in which directions.
+	 */
+	resize?: 'both' | 'vertical' | 'horizontal' | 'none'
+}>(), {
+	helperText: undefined,
+	inputClass: '',
+	label: undefined,
+	placeholder: undefined,
+	resize: 'both',
+})
+
+/**
+ * The value of the text area
+ */
+const modelValue = defineModel<string>({ required: true })
+
+/**
+ * The native text area component instance
+ */
+const textAreaElement = useTemplateRef('input')
+
+const componentId = createElementId()
+
+// needs to be a getter as attrs are not reactive
+const attrs = useAttrs()
+const inputId = () => String(attrs.id) || componentId
```

Comment:
Using a specific attribute is component's logic (not just inherit all the attrs on a child node) is a good sign that it must have been a prop, not an attr.

Then there are no problems with types, reactivity and it manages the component explicitly

Reply:
Makes sense, pushed a cleanup of this